ddtrace/tracer/spancontext.go

```diff
@@ -205,7 +209,28 @@ func (t *trace) drop() {
 	atomic.CompareAndSwapInt64((*int64)(&t.samplingDecision), int64(decisionNone), int64(decisionDrop))
 }
 
-func (t *trace) setSamplingPriorityLocked(p float64) {
+func (t *trace) setTag(key, value string) {
+	if t.tags == nil {
+		t.tags = make(map[string]string, 1)
+	}
+	t.tags[key] = value
+}
+
+func (t *trace) buildNewUpstreamServices(service string, priority int, sampler samplerName, rate float64) string {
```

Comment:
Can this be a simple function and not a method?
```suggestion
func compactUpstreamServices(service string, priority int, sampler samplerName, rate float64) string {
```

You would then set it as `t.upstreamServices + compactUpstreamService(...)`

Reply:
+1 to Gabriel's comment. Pass in the existing services if necessary for the `;` logic. There's no reason to bind this function to a `*trace`.